dissent-oracle/README.md

```diff
@@ -1,9 +1,47 @@
-# Dissent Oracle
+# Dissent Oracle <img align="right" src="https://github.com/1Hive/website/blob/master/website/static/img/bee.png" height="80px" />
 
-Oracle implementation installed along with the Dandelion Voting app to prevent engagement when a user has recently voted in favour of a proposal.
+The Dissent Oracle app is an [ACL Oracle](https://hack.aragon.org/docs/acl_IACLOracle). ACL Oracles are small helper functions that plug in to Aragon's access control list (ACL) to do more sophisticated permission evaluation. The Dissent Oracle is installed along with the Dandelion Voting app to prevent engagement when a user has recently voted in favour of a proposal.
 
-## Usage
+#### 🐲 Project stage: development
 
-Applied to an Aragon app's permission's params.
+The Time Lock app is still in development, a first implementation was published to `time-lock.open.aragonpm.eth`. If you are interested in contributing please see our open [issues](https://github.com/1hive/time-lock-app/issues).
+
+#### 🚨 Security review status: pre-audit
+
+The code in this repo has not been audited.
+
+## How does it work?
+
+The Dissent Oracle is initialized with a dissent window. The dissent window is the window of time in which an account cannot dissent (exit the Dandelion Org) if they have performed an action (such as voting `yes` for a proposal). The Dandelion Voting app queries the Dissent Oracle to check if an account voted `yes` within the dissent window. In the context of Dandelion Orgs this is used to check if an account can or cannot redeem tokens from the Redemptions app.
+
+> Note: this will not work with the regular Aragon voting application because it does not track the last block in which a voter voted yea.
+
+## Initialization
+
+The Dissent Oracle app is initialized with an `address _dandelionVoting` and `uint64 _dissentWindowBlocks`. The `address _dandelionVoting` is the address of the voting app that the Dissent Oracle is to query. The `uint64 _dissentWindowBlocks` is the time window within which the dissent oracle should return a boolean.
```

Comment:
> is the time window within

Probably should say is the time window (expressed as a number of blocks) within which the dissent oracle should return a boolean.